dynawo/sources/Solvers/VariableTimeStep/SolverIDA/DYNSolverIDA.cpp

```diff
@@ -699,6 +820,40 @@ SolverIDA::solveStep(double tAim, double& tNxt) {
     case IDA_TSTOP_RETURN:
       msg = "IDA_TSTOP_RETURN";
       break;
+    case IDA_CONV_FAIL:
+      if (countForceReinit_ == 0) {
+        Trace::info() << DYNLog(SolverIDAForceReinitConvFail) << Trace::endline;
+        int msbsetAlgJSave = msbsetAlgJ_;
+        modeChangeType_t minimumModeChangeTypeForAlgebraicRestorationSave = minimumModeChangeTypeForAlgebraicRestoration_;
+        msbsetAlgJ_ = 1;
+        minimumModeChangeTypeForAlgebraicRestoration_ = ALGEBRAIC_J_J_UPDATE_MODE;
+        model_->setModeChangeType(minimumModeChangeTypeForAlgebraicRestoration_);
```

Comment:
this modification is never reverted, is it normal?

Reply:
no I updated this